tools/codegen/api/types.py

```diff
@@ -227,6 +211,36 @@ def name(self) -> ArgName:
     TupleCType
 ]
 
+# A NamedCType is short for Named C++ semantic type.  A NamedCType represents a C++ type, plus
+# semantic information about what it represents.  For example, consider the
+# argument "bool pin_memory"; its normal C++ type is "bool", but its C++
+# semantic type also keeps track that this represents a "pin_memory"; you can't
+# just use a random other boolean in a context where you need a "pin_memory"!
+#
+# NamedCTypes encode C++ type structure as needed for translation.  Right now we
+# track references and optional, but don't, for example, track ArrayRef.  If
+# you need trnsnlations that know about these types, beef up this data
+# structure.
```

Comment:
I think this comment block actually still belongs with `CType`, no? (Also, the part about `ArrayRef` is out of date I think)